### PR TITLE
BOM-2930: Replace Travis with GitHub CI

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-2930](https://openedx.atlassian.net/browse/BOM-2930)

### Description
- Replace `Travis` with `GitHub CI`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: pip install -r requirements/ci.txt
+
+    - name: Run Tests
+      run: make quality
+
+    - name: Run Coverage
+      if: matrix.python-version == '3.8' && matrix.toxenv=='django32'
```
Comment (nedbat) [inline .github/workflows/ci.yml:37]: Does the matrix have a `toxenv` setting?

Comment (UsamaSadiq) [reply .github/workflows/ci.yml]: Removed the unnecessary condition.

**Comment by nedbat** [comment]
Now it looks like the entire "Run Coverage" step is gone? Though none of the force-pushes seems to have done it? I'm confused what happened.

**Comment by UsamaSadiq** [comment]
After checking on the codecov site and comparing with existing Travis, I found that the coverage wasn't already working on this repo. Since there aren't any unit tests running in the CI so I removed the coverage step altogether.

**Review by nedbat — APPROVED** [review]

**Pull request merged**